qemu/tests/live_snapshot_runtime.py

```diff
@@ -0,0 +1,65 @@
+import time
+import random
+
+from autotest.client.shared import error
+from autotest.client.shared import utils
+from qemu.tests import live_snapshot_basic
+
+
+class LiveSnapshotRuntime(live_snapshot_basic.LiveSnapshot):
+
+    process = []
+
+    def __init__(self, test, params, env, tag):
+        super(LiveSnapshotRuntime, self).__init__(test, params, env, tag)
+
+    @error.context_aware
+    def reboot(self):
+        """
+        Reset guest with system_reset;
+        """
+        method = self.params.get("reboot_method", "system_reset")
+        return super(LiveSnapshotRuntime, self).reboot(method=method)
+
+    def action_before_start(self):
+        """
+        start pre-action in new threads;
+        """
+        for param in self.params.get("before_start").split():
+            if hasattr(self, param):
+                fun = getattr(self, param)
+                bg = utils.InterruptedThread(fun)
+                bg.start()
+                if bg.isAlive():
+                    self.process.append(bg)
+            time.sleep(random.randint(0, 50))
+
+    def clean(self):
+        """
+        clean up sub-process and trash files
+        """
+        for bg in self.process:
+            bg.join()
+        super(LiveSnapshotRuntime, self).clean()
+
+
+def run(test, params, env):
+    """
+    live_snapshot runtime test:
+    1). boot guest;
+    2). do some operations, like reboot;
+    3). create snapshots during the operations in step 2;
+    4). waiting operations done and check guest is alive;
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    tag = params.get("source_image", "image1")
+    runtime_test = LiveSnapshotRuntime(test, params, env, tag)
+    try:
```

Comment:
Thanks for your suggestions.

Those actions have already covered exception handling in the func().
I have updated finally part to avoid true error reasons will be covered.

What do you think?